components/timezone/src/ids.rs

```diff
@@ -248,37 +253,38 @@ impl TimeZoneIdMapperBorrowed<'_> {
     /// let mapper = TimeZoneIdMapper::new();
     /// let mapper = mapper.as_borrowed();
     ///
-    /// let result = mapper.canonicalize_iana("Asia/CALCUTTA").unwrap();
+    /// let result = mapper.canonicalize_iana("Asia/CALCUTTA");
     ///
     /// assert_eq!(result.0, "Asia/Kolkata");
     /// assert!(matches!(result.0, Cow::Owned(_)));
     /// assert_eq!(*result.1, "inccu");
     ///
     /// // Borrows when able:
-    /// let result = mapper.canonicalize_iana("America/Chicago").unwrap();
+    /// let result = mapper.canonicalize_iana("America/Chicago");
     /// assert_eq!(result.0, "America/Chicago");
     /// assert!(matches!(result.0, Cow::Borrowed(_)));
     ///
     /// // Unknown IANA time zone ID:
-    /// assert_eq!(mapper.canonicalize_iana("America/San_Francisco"), None);
+    /// assert_eq!(mapper.canonicalize_iana("America/San_Francisco").1, TimeZoneBcp47Id::unknown());
```

Comment:
Nit: Also show the equivalence to "Etc/Unknown"

Reply:
I've made this fallible again, because `Etc/Unknown` is not a canonical zone (it's not an IANA zone at all).